source/core/repository/repository.ts

```diff
@@ -576,16 +583,24 @@ export class Repository<TModel = unknown> {
                             const func = keysFunc[operator as keyof WhereClause];
                             func(query, key, opValue);
                         }
-                } else if (key === '$q') {
+                } else if (key === '$q' && typeof value === 'string') {
                     for (const field of this._table.fields)
-                        query.where(field, 'like', `%${value as string}%`);
+                        if (checkField(field))
```

Comment:
Cette condition est inutile puisque tu fais déjà un for of sur this._table.fields